### PR TITLE
perf(app_user): replace CircleAvatar+NetworkImage with MinglitAvatarImage

### LINKED ISSUE #1936: [Perf] NetworkImage used directly for CircleAvatar in multiple places — bypasses MinglitImage caching
## Context
Five call sites use `NetworkImage(url)` directly inside `CircleAvatar.backgroundImage` instead of going through `MinglitImage` or a caching-aware variant:

| File | Line | Description |
|------|------|-------------|
| `apps/app_user/lib/src/features/event/detail/event_detail_content.dart` | 305 | Partner profile avatar in event detail |
| `apps/app_user/lib/src/features/home/my_page.dart` | 280 | Profile tile avatar |
| `apps/app_user/lib/src/features/settings/blocked_partners_page.dart` | 92 | Blocked partner avatar |
| `apps/app_user/lib/src/features/home/home_page.dart` | 133 | App bar user avatar |
| `apps/app_user/lib/src/features/home/widgets/event_now_phases/results_content.dart` | 146 | Match result partner avatar |

`MinglitImage` uses `cached_network_image` under the hood with Supabase image transform support. Direct `NetworkImage` usage skips:
- Memory and disk caching (re-downloads on every rebuild/scroll)
- Supabase image transform (`width` param for bandwidth optimization)
- The `onBackgroundImageError` fallback (only `home_page.dart` bothers to add it; the others have no error handling)

## Why this matters
- **Performance**: Small avatars (24–48px radius) loading from full-resolution URLs on every scroll rebuild waste bandwidth and cause jank during list scrolling.
- **UX**: Without `onBackgroundImageError`, a broken image URL silently renders as empty `CircleAvatar` with no fallback icon visible.
- **Maintainability**: `MinglitImage` is the single surface to update when Supabase URL transform policy changes (like the recent #1918 hotfix). Direct `NetworkImage` use requires per-call-site fixes.

## Acceptance Criteria
- [ ] Replace each `NetworkImage(url)` `CircleAvatar` with a `MinglitImage`-backed avatar widget (e.g. a shared `MinglitAvatarImage(url: ..., radius: ...)` wrapper, or use `MinglitImage` with a `BoxDecoration`).
- [ ] Ensure all 5 call sites have an error/null fallback icon.
- [ ] `cacheWidth` is set appropriately for the rendered size (e.g. `(radius * 2 * devicePixelRatio).round()`).

## Effort
1h

### PR DESCRIPTION
**Pull request by Mark-Yun**
Scheduler: needs-swe-sonnet-1

Closes #1936

## UX 가이드 (UI 변경 게이트)

ux-designer (`needs-uiux-claude-1`) 승인 가이드: https://github.com/Mark-Yun/minglit/issues/1936#issuecomment-4333811199

- 로딩 Shimmer: 디자인 시스템 표준 통일 → ✅ 개선
- 에러 Fallback Icon 가시화: Empty/Error State 가시성 원칙에 부합 → ✅ 의도된 UX
- 5개 콜사이트 아이콘 크기: 원본 시각 유지, 4/5 토큰 사용 (1건은 후속 정리 메모로 분리)

## Summary

- 5곳에서 `CircleAvatar(backgroundImage: NetworkImage(url))`를 직접 사용하던 코드를 `MinglitAvatarImage`로 교체
- `NetworkImage` 직접 사용은 ResizeImage 최적화 없이 풀해상도 디코딩, 에러 핸들링 없음
- `MinglitAvatarImage` 신규 위젯 도입: `ResizeImage(cacheWidth = radius*2*dpr)` + `errorBuilder`로 fallback 아이콘 표시 + `ClipOval`로 원형 클리핑

## 변경 대상 (5곳)

| 파일 | radius | fallbackIcon |
|------|--------|-------------|
| `event_detail_content.dart` | 12 | `Icons.store` |
| `home_page.dart` | 14 | `Icons.person` |
| `my_page.dart` | 24 | `Icons.person` (branded colours) |
| `blocked_partners_page.dart` | 20 | `Icons.store` |
| `match_results_content.dart` | 24 | `Icons.person` (branded colours) |

## Architecture

```
MinglitAvatarImage
  url == null → CircleAvatar(fallbackIcon)
  url != null → ClipOval > Image(ResizeImage(NetworkImage(url)))
                  loadingBuilder → CircleAvatar placeholder
                  errorBuilder   → CircleAvatar(fallbackIcon)
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    if (url == null || url!.isEmpty) return placeholder;
+
+    // Fix #1936: decode at physical-pixel size to avoid full-res decode on scroll
+    final dpr = MediaQuery.devicePixelRatioOf(context);
+    final cacheW = (size * dpr).round();
+
+    return ClipOval(
+      child: Image(
+        image: ResizeImage.resizeIfNeeded(cacheW, cacheW, NetworkImage(url!)),
```
Comment (onlyhyeok-cmd) [inline apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: Scheduler: needs-review-codex-1

`MinglitAvatarImage` still builds directly on `NetworkImage`, so this PR reintroduces a second image-loading primitive instead of routing avatars through the shared image surface. The original issue was not just the five call sites; it was that avatar rendering bypassed the central place where width/error/loading/semantics policy is maintained. After this lands, any future change to `MinglitImage` behavior still requires a separate fix here, which puts us back into the same maintenance trap #1936 was trying to remove.

Please make the avatar wrapper delegate to `MinglitImage` (or move a shared circular-avatar primitive next to it in the shared package) rather than reimplementing `ResizeImage(NetworkImage(...))` locally.

**Review by onlyhyeok-cmd — CHANGES_REQUESTED** [review]
Scheduler: needs-review-codex-1

Request changes on commit `078780bd57d98b74594defa5c63acd0f977093f5`.

The new avatar wrapper centralizes the five call sites, but it still bypasses the shared image primitive by constructing `ResizeImage(NetworkImage(...))` directly. That leaves image policy split across `MinglitImage` and `MinglitAvatarImage`, so the next loading/error/transform/semantics fix has to be patched twice. Issue #1936 was specifically about removing that bypass, not just moving it into a different file.

Please rework the wrapper so it delegates to `MinglitImage` or lives next to it as a shared primitive, then re-request review.

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  final IconData fallbackIcon;
+  final Color? backgroundColor;
+  final Color? fallbackIconColor;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = radius * 2;
+
+    final Widget placeholder = CircleAvatar(
+      radius: radius,
```
Comment (onlyhyeok-cmd) [inline apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: **[버그] 폴백 아이콘 사이즈 회귀 — 두 call site에서 원본과 달라짐**

`size: radius`로 고정하면 각 call site의 원본 아이콘 크기를 깨뜨린다:

| call site | 원본 size | 현재 size |
|---|---|---|
| `event_detail_content.dart` | `MinglitIconSize.xsmall` = **16px** | `radius` = **12px** (-25%) |
| `blocked_partners_page.dart` | Flutter 기본값 **24px** | `radius` = **20px** (-17%) |

`home_page.dart` (radius=14, 원본=14)와 `my_page.dart`/`match_results_content.dart` (radius=24, 원본=24)는 우연히 일치해서 문제없음.

**수정안**: `fallbackIconSize` 파라미터 추가 (기본값 `null` → Flutter 기본 사용):

```dart
final double? fallbackIconSize;
// ...
child: Icon(fallbackIcon, color: fallbackIconColor, size: fallbackIconSize),
```

`event_detail_content.dart` call site → `fallbackIconSize: MinglitIconSize.xsmall` 추가로 복원.

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+
+/// Circular avatar widget backed by a cached, resize-optimised network image.
+///
+/// Replaces direct `CircleAvatar(backgroundImage: NetworkImage(url))` usage,
+/// which skips ResizeImage cache-width optimisation and has no error handling.
+///
+/// Shows [fallbackIcon] in a plain `CircleAvatar` when [url] is null, while
+/// the image is loading, or when the image fails to load.
+class MinglitAvatarImage extends StatelessWidget {
```
Comment (onlyhyeok-cmd) [inline apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: **[테스트 부재] 신규 공통 위젯에 위젯 테스트 없음**

5곳에서 사용하는 공통 위젯인데 테스트가 전혀 없다. 지금 막히지 않는 회귀:
- `errorBuilder` 가 제거되면 이미지 로드 실패 시 폴백 없이 빈 이미지 렌더링
- `url == null || url!.isEmpty` 가드가 깨지면 빈 URL로 NetworkImage 호출 → 무한 에러

`test/common/widgets/minglit_avatar_image_test.dart` 추가, 최소 3 케이스:
1. `url == null` → `CircleAvatar` + fallback icon 렌더링
2. 유효 url → `ClipOval > Image` 렌더링
3. image load 실패 → `errorBuilder` 경로로 fallback icon 렌더링

**Review by onlyhyeok-cmd — CHANGES_REQUESTED** [review]
Reviewed at 078780b.

두 가지 수정 필요. 나머지 구현(ResizeImage 최적화, errorBuilder, ClipOval 교체)은 의도대로이고 문제없음.

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart (hunk 1 above)
Comment (Mark-Yun) [reply apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: Fixed in commit c4bf291. MinglitAvatarImage is now in shared/packages/mds/core alongside MinglitImage. The URL path delegates to ClipOval(child: MinglitImage(…)) — no more ResizeImage/NetworkImage duplication. Null/empty URL still shows CircleAvatar with icon fallback.

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart (hunk 2 above)
Comment (Mark-Yun) [reply apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: Fixed in commit c4bf291. Added fallbackIconSize param (default null → Flutter 24px). event_detail_content.dart now passes fallbackIconSize: MinglitIconSize.xsmall (16px), restoring the original. home_page.dart passes fallbackIconSize: 14 to preserve its 14px behavior. blocked_partners_page.dart uses null default which restores the original Flutter 24px default.

File: apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart (hunk 3 above)
Comment (Mark-Yun) [reply apps/app_user/lib/src/common/widgets/minglit_avatar_image.dart]: Added apps/app_user/test/src/common/widgets/minglit_avatar_image_test.dart (commit c4bf291) with 5 test cases: null URL → CircleAvatar, empty URL → CircleAvatar, non-null URL → ClipOval, image load failure (network fails in flutter_test by default) → no exception, fallbackIconSize override → correct icon.size.

**Comment by Mark-Yun** [comment]
All three review points addressed (commits bef0222, c4bf291):

1. **Delegation to MinglitImage**: Widget moved to `shared/packages/mds/core` alongside `MinglitImage`. URL path is now `ClipOval(child: MinglitImage(…))` — `ResizeImage`/`NetworkImage` construction is no longer duplicated.
2. **Icon size regression**: Added `fallbackIconSize` param (default `null` → Flutter 24px). `event_detail_content.dart` passes `MinglitIconSize.xsmall`, `home_page.dart` passes `14`, others use default.
3. **Tests**: `apps/app_user/test/src/common/widgets/minglit_avatar_image_test.dart` — 5 cases covering null/empty URL, network URL, load failure, and `fallbackIconSize` override.

**Review by onlyhyeok-cmd — DISMISSED** [review]
Scheduler: needs-review-sonnet-1

All three concerns from the previous review are resolved:

1. **MinglitAvatarImage delegated to MinglitImage**: Widget now lives in `shared/packages/mds/core` and routes network images through `ClipOval(child: MinglitImage(...))`. No more duplicate `ResizeImage(NetworkImage(...))` primitive. A single MinglitImage fix now covers avatar rendering too.

2. **Fallback icon size regression fixed**: `fallbackIconSize` param added (default `null` → Flutter 24px). `event_detail_content.dart` passes `MinglitIconSize.xsmall` (16px) and `home_page.dart` passes `14` — both restored to original sizes. `blocked_partners_page.dart` uses null default which matches the original Flutter 24px.

3. **Widget tests added**: 5 cases in `minglit_avatar_image_test.dart` covering null URL, empty URL, valid URL → ClipOval, image load failure (no exception), and fallbackIconSize override.

Reviewed at commit c7a31ef.

**Comment by Mark-Yun** [comment]
🤖 **needs-swe-sonnet-1**: CI 실패 원인 분석 완료.

브랜치가 PR #2036 (event_repository 분리 리팩토링)이 dev에 머지되기 이전 커밋을 기반으로 생성되어, 무관한 파일들이 diff에 포함됐습니다. 이로 인해 purchase history 관련 테스트들이 잘못된 코드 상태에서 실행돼 실패했습니다.

**대응**: `origin/dev`로 리베이스하여 불필요한 diff를 제거했습니다. PR diff가 이제 #1936 관련 파일만 포함합니다:
- `MinglitAvatarImage` 위젯 (MDS 패키지)
- 5개 call-site 교체 (app_user)
- 회귀 방지 테스트 + 골든 이미지 업데이트

CI가 재실행 중입니다. 통과하면 auto-merge됩니다.

**Review by onlyhyeok-cmd — CHANGES_REQUESTED** [review]
Reviewed at commit a3477d096913df22840e78ec9fba4fd15a2af5f4

## UI 변경 게이트 미통과 (request-changes)

골든 테스트 이미지 2개가 변경됨:
- `test/alchemist/goldens/ci/blocked_partners_with_data.png`
- `test/alchemist/goldens/ci/event_detail_page_states.png`

이는 시각 출력이 실제로 바뀌었음을 확인해준다. 구체적으로:
- **로딩 상태**: `CircleAvatar` placeholder → `MinglitImage` shimmer/placeholder로 전환
- **에러 상태**: silent empty circle → 폴백 아이콘 가시화

팀 UI 변경 게이트 기준에 따르면, 화면·위젯 교체가 있을 때 **ux-designer 가이드 인용(이슈/코멘트 링크)**이 PR 본문에 있어야 머지 가능하다. PR 본문에 해당 인용이 없다.

### 대응 방법

1. 이슈 #1936에 `needs-uiux` 라벨 추가 → ux-designer 가이드 수령
2. PR 본문에 해당 가이드 링크 인용 후 push
3. `needs-review` 라벨 다시 부여

---

**코드 품질 자체는 이상 없다.** 위젯 구현 로직, 5개 콜사이트 교체, 5개 테스트 케이스 모두 수용 가능하다. 게이트 절차만 통과되면 approve 가능.

**Comment by Mark-Yun** [comment]
🤖 **needs-uiux-gemini-1** 가이드 작성 완료 및 게이트 통과 알림.

이슈 #1936에 UX 가이드를 작성했습니다.
- [가이드 링크](https://github.com/Mark-Yun/minglit/issues/1936#issuecomment-4333705746)

시각적 변경 사항(Shimmer, 폴백 아이콘)이 디자인 시스템에 부합함을 확인하여 승인했습니다.  라벨을 다시 부여했으니 재검토 부탁드립니다. @onlyhyeok-cmd 님.

**Review by onlyhyeok-cmd — CHANGES_REQUESTED** [review]
Scheduler: needs-review-codex-1

Request changes on commit `a3477d096913df22840e78ec9fba4fd15a2af5f4`.

1. `shared/packages/mds/core/lib/src/ui/widgets/common/minglit_avatar_image.dart:45` now hands every non-empty URL to `MinglitImage`, whose `errorBuilder` is hard-coded to `Icons.image_not_supported_outlined` in `shared/packages/mds/core/lib/src/ui/widgets/common/minglit_image.dart:134`. That means a broken partner avatar no longer shows the caller-provided `fallbackIcon`/`fallbackIconColor`; `Icons.store` and the branded person icon are lost exactly in the load-failure path this PR claims to fix. Issue #1936 and the UX guidance on #1936 both require the explicit fallback icon to remain visible for error states, so the wrapper needs to preserve caller-specific fallback rendering for failed network loads as well, not just for null/empty URLs.

2. The new widget test file only checks that the failed network load does not throw (`apps/app_user/test/src/common/widgets/minglit_avatar_image_test.dart:63`), but it never asserts which placeholder is shown after failure. That gap is why the regression above slipped through. Please add a regression test that pumps a non-empty URL and verifies the same `fallbackIcon`/styling is rendered after the image load fails.

3. The UI gate is still not satisfied in the PR description. The approved UX comment on issue #1936 explicitly says to cite that guide link in the PR body, but `gh pr view 2035 --json body` still shows no such citation. Per the team rule for UI-changing PRs, the link must be in the PR description before merge.

**Comment by Mark-Yun** [comment]
🤖 **needs-uiux-claude-1** UX 가이드 (상세본) 작성 완료.

리뷰어 추가 요청 3가지에 대한 디자인 시스템 기준 검증을 수행했습니다.
- 가이드: https://github.com/Mark-Yun/minglit/issues/1936#issuecomment-4333811199
- PR description에 인용 추가, `needs-review` 라벨 재부여

**판정 요약**: ✅ 승인
- 로딩 Shimmer: 디자인 시스템 표준 통일 (개선)
- 에러 Fallback Icon: Empty/Error State 가시성 원칙 부합 (의도된 UX)
- 5개 콜사이트 아이콘 크기: 4/5 토큰 사용, 1건(`home_page.dart` `fallbackIconSize: 14`)은 후속 토큰화 메모로 분리 — 이번 PR 머지 차단하지 않음

@onlyhyeok-cmd 재검토 부탁드립니다.

**Review by onlyhyeok-cmd — CHANGES_REQUESTED** [review]
Scheduler: needs-review-codex-1

Request changes on commit `a3477d096913df22840e78ec9fba4fd15a2af5f4`.

UI gate citation is now present in the PR body, so that blocker is resolved. Two code-level blockers remain:

1. `shared/packages/mds/core/lib/src/ui/widgets/common/minglit_avatar_image.dart` still sends every non-empty URL through `MinglitImage`, whose `errorBuilder` is fixed to `Icons.image_not_supported_outlined` in `shared/packages/mds/core/lib/src/ui/widgets/common/minglit_image.dart`. That means the broken-image path no longer preserves the caller-provided `fallbackIcon` / `fallbackIconColor` for partner avatars and branded user avatars. Issue #1936 acceptance criteria and the UX guidance on #1936 both require those explicit per-call-site fallbacks to remain visible for load failures, not only for null or empty URLs.

2. `apps/app_user/test/src/common/widgets/minglit_avatar_image_test.dart` still does not assert the failed-network state renders the caller-specific fallback. The current failure test only checks that no exception is thrown, so it would keep passing while the broken-image UI regressed exactly as it has here. Add a regression test that pumps a non-empty URL, lets the image fail, and verifies the expected `fallbackIcon` (and color/styling where applicable) is what the user sees after the failure.

Please fix those two points and re-request review.

**Comment by Mark-Yun** [comment]
🤖 **needs-swe-sonnet-1** 수정 완료.

**수정 내용:**

1. **fallback icon 회귀 수정** — `MinglitImage`에 optional `errorWidget` 파라미터 추가. `MinglitAvatarImage`는 이제 에러 시 caller의 `fallbackIcon`/`fallbackIconColor`로 구성된 `CircleAvatar`를 `errorWidget`으로 전달 → `Icons.image_not_supported_outlined` 대신 branded icon 표시.

2. **테스트 assertion 추가** — "image load failure renders without throwing" 테스트를 "image load failure shows caller fallbackIcon, not generic icon"으로 교체. 이제 `Icons.store` 아이콘이 렌더링되고 `Icons.image_not_supported_outlined`이 나타나지 않음을 명시적으로 assert.

commits: 438cdbc5, e2c401c7

**Review by onlyhyeok-cmd — DISMISSED** [review]
Scheduler: needs-review-codex-1

Approved at commit `ec13495e70e699c44aafff0dc908f621177c6f36`.

Previous blockers are resolved:
- `MinglitAvatarImage` now preserves caller-specific fallback rendering on failed network loads via `MinglitImage(errorWidget: ...)`.
- The regression test now asserts the failed-load path shows the caller fallback icon rather than the generic broken-image icon.
- The PR body includes the required UX guide citation for the visual change.

I could not run `flutter test` locally in this environment because `flutter` is not installed here. CI checks were still in progress when I reviewed; merge should still wait on those checks.
